instance/tasks.py

```diff
@@ -76,47 +74,81 @@ def start_deployment(
     logger.info('Retrieving deployment: ID=%s', instance_ref_id)
     deployment = OpenEdXDeployment.objects.get(pk=deployment_id)
 
+    # For redundancy - if app server deployment was successful even after cancelling, we prevent the change in
+    # active appserver here
     if deployment.cancelled:
         logger.info('Deployment %s was cancelled, returning.', deployment.id)
-        return False
+        return
 
     logger.info('Spawning servers for deployment %s [%s]', deployment, deployment.id)
+    old_server_ids = list(instance.appserver_set.filter(_is_active=True).values_list('id', flat=True))
```

Comment:
@Kelketek, since `old_server_ids` is only used by the `check_deactivation()` method and it already has access to the `instance_ref_id`, can't we avoid getting it here and passing it two levels down?